R/render.R

```diff
@@ -1,15 +1,28 @@
-epi_from_subpopulation <- function(target, age, lower, upper) {
+count_of_subpopulation <- function(target, age, lower, upper) {
   in_range <- individual::Bitset$new(target$max_size)$insert(
     which((age >= lower) & (age <= upper))
   )
+  n_in_range <- in_range$size()
+  if (n_in_range == 0) {
+    return(0)
+  }
+  n_in_range
+}
 
+count_of_detected <- function(target, age, lower, upper) {
+  in_range <- individual::Bitset$new(target$max_size)$insert(
+    which((age >= lower) & (age <= upper))
+  )
```

Comment:
We calculate this twice every timestep now. Once for `count_of_subpopulation` and once for `count_of_detected`. Is there an easy way to calculate it once?

Reply:
I'm trying to combine these into one function which outputs a list with two results (numerator, denominator). Is this what you had in mind? It is failing on the tests so far, "attempt to apply non-function", but could be cool if we get it to work.

count_of_detected <- function(target, age, lower, upper){
  in_range <- individual::Bitset$new(target$max_size)$insert(
    which((age >= lower) & (age <= upper))
  )
  
  denominator <- in_range$size()
  if (denominator == 0) {
    return(0)
  }  
  
  numerator <- in_range$and(target)$size()
  
  result <- c(numerator, denominator)

  result
}

p <- count_of_detected( 
        detected,
        age,
        lower,
        upper
      )
      renderer$render(paste0('n_detect_', lower, '_', upper), p[[1]], timestep)